app/src/main/java/com/example/socialmediavbsanalay/data/dataSourceImpl/gallery/GalleryDataSourceImpl.kt

```diff
@@ -0,0 +1,72 @@
+package com.example.socialmediavbsanalay.data.dataSourceImpl.gallery
+
+
+import android.content.ContentUris
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+import android.provider.MediaStore
+import android.util.Log
+import com.example.socialmediavbsanalay.data.dataSource.gallery.GalleryDataSource
+import com.example.socialmediavbsanalay.domain.model.Gallery
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+class GalleryDataSourceImpl @Inject constructor(
+    private val context: Context
```

Comment:
Burada contexti kendimiz passlemek yerine aşağıdaki dagger-hilt yapısını kullanabilir misin?

@ApplicationContext private val context: Context

bu kod sayesinde hilt contexti kendi inject edecektir ve applicationContext kullanılacağı için potansiyel memory leak riski ortadan kalkacaktır.

Reply:
Galerylerle ilgili düzenleme kısımlarıyla ileride ilgileneceğim